test/Microsoft.AspNet.BasicMiddleware.Tests/OverrideHeaderMIddlewareTest.cs

```diff
@@ -0,0 +1,208 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.TestHost;
+using Xunit;
+
+namespace Microsoft.AspNet.BasicMiddleware
+{
+    public class OverrideMiddlewareHeaderTests
+    {
+        [Fact]
+        public async Task OverrideHeaderMiddleware_XForwardEnabled()
+        {
+            var options = new OverrideHeaderMiddlewareOptions(false);
+            options.XForwardedForEnabled = true;
+            TestServer server = TestServer.Create(app =>
+            {
+                app.UseOverrideHeaders(options);
+                app.Run(context =>
+                {
+                    Assert.Equal("11.111.111.11", context.Connection.RemoteIpAddress.ToString());
+                    return Task.FromResult(0);
+                });
+            });
+
+            var req = new HttpRequestMessage(HttpMethod.Get, "XForwardedFor enabled");
+            req.Headers.Add("X-Forwarded-For", "11.111.111.11");
+            HttpResponseMessage resp = await server.CreateClient().SendAsync(req);
+        }
+
+        [Fact]
+        public async Task OverrideHeaderMiddleware_XForwardedHostEnabled()
+        {
+            var options = new OverrideHeaderMiddlewareOptions();
+            TestServer server = TestServer.Create(app =>
+            {
+                app.UseOverrideHeaders(options);
+                app.Run(context =>
+                {
+                    Assert.Equal("testHost", context.Request.Host.ToString());
+                    return Task.FromResult(0);
+                });
+            });
+
+            var req = new HttpRequestMessage(HttpMethod.Get, "X-Forwarded-For Enabled");
+            req.Headers.Add("X-Forwarded-Host", "testHost");
+            HttpResponseMessage resp = await server.CreateClient().SendAsync(req);
```

Comment:
And in a case, some how a bug causes the line of code on _43_ is not executed, how to capture this issue?